apps/system/lockscreen/js/lockscreen_inputpad.js

```diff
@@ -163,7 +167,19 @@
       var span = this.passcodeCode.childNodes[i];
       if (span) {
         if (this.states.passCodeEntered.length > i) {
+        var value = this.states.passCodeEntered.substr(i, this.states.passCodeEntered.length);
+        if ((this.states.passCodeEntered.length - 1 === i) && (keyCode && (keyCode !== 'b'))) {
+          (function(aspan, currentValue) {
+            aspan.textContent = currentValue;
+            setTimeout(function() {
+              aspan.textContent = '';
+              aspan.dataset.dot = true;
+            }, CODE_VISIBLE_TIMEOUT);
```

Comment:
I suspect that 1 second will be too long for this, so it is good that it is in a constant that is easy to change.

I think that we also need to add code to ensure that if the user is entering their pin quickly, all previous digits are replaced with dots as soon as a new key is typed. There should never be more than one digit displayed at a time, but with the current code it looks like there can be.